### ISSUE TITLE
Prometheus operator helm chart fun and games

### ISSUE THREAD

**Issue by gitfool** [pulumi/pulumi-kubernetes#824]
I've been struggling to install the [prometheus-operator](https://github.com/helm/charts/tree/master/stable/prometheus-operator) to an EKS cluster via the helm chart.

I wanted everything to be tucked into a "monitoring" namespace, and I eventually worked out what I needed to disable, which makes sense in hindsight given the EKS control plane, and now have the following:
```ts
const monitoringNamespace = new k8s.core.v1.Namespace("monitoring", { metadata: { name: "monitoring" } }, { provider: provider });

function fixGrafanaTest(obj: any) {
    if (obj.metadata.name === "po-grafana-test") {
        if (obj.kind === "Pod") {
            obj.metadata.annotations = {
                "pulumi.com/skipAwait": "true"
            };
        }
    }
}

function setMonitoringNamespace(obj: any) {
    if (obj.metadata.namespace === undefined) {
        obj.metadata.namespace = "monitoring";
    }
}

const prometheusOperatorChart = new k8s.helm.v2.Chart("po", {
    repo: "stable",
    chart: "prometheus-operator",
    version: "6.11.0",
    namespace: "monitoring",
    transformations: [ fixGrafanaTest, setMonitoringNamespace ],
    values: {
        kubeControllerManager: { enabled: false },
        kubeEtcd: { enabled: false },
        kubeScheduler: { enabled: false },
        kubeTargetVersionOverride: k8sVersion,
        prometheusOperator: { createCustomResource: false }
    }
}, {
    dependsOn: monitoringNamespace,
    provider: provider
});
```
I'd like highlight the following issues:
- If the `prometheusOperatorChart` referenced `monitoringNamespace` using `namespace: monitoringNamespace.metadata.name`, then I'd get a warning about `[Can't preview] all chart values must be known ahead of time to generate an accurate preview`, so I'm using the same constant value instead. Maybe this could be improved? (https://github.com/pulumi/pulumi/pull/3245).
- The Grafana test is using a helm hook on [test-success](https://github.com/helm/charts/blob/master/stable/grafana/templates/tests/test.yaml#L11) and is always failing its [grafana health check](https://github.com/helm/charts/blob/master/stable/grafana/templates/tests/test-configmap.yaml#L12-L17), which I initially thought was due to the timing of helm hooks not being supported when using `helm template`, like Pulumi does behind the scenes, but then I'd expect subsequent runs to succeed since Grafana is running then, but it _always_ fails so probably something more subtle with the test itself.
  - Subsequently there is the problem that Pulumi waits until it times out; handily I could add an annotation to "skipAwait", which works around my impatience, but then it seems to me that Pulumi could be smarter here since the pod specifies to [never restart](https://github.com/helm/charts/blob/master/stable/grafana/templates/tests/test.yaml#L64), so it should stop waiting after the first failure.
  - So now it fails fast, but I _still_ need a workaround to avoid the failing test, like using another transformation to neuter or preferably remove it altogether, along with all of its [supporting resource](https://github.com/helm/charts/tree/master/stable/grafana/templates/tests) detritus. I can see how to neuter the pod by modifying its image etc, but not so much the supporting resources.

**Comment by gitfool** [comment]
@hausdorff mentioned in Slack a possible hack to remove resources in a transformation:
> if you set apiVersion: "v1" and kind: "List" I think it should work

I'll give that a go in the meantime, but it would obviously be better to have a first-class way to remove resources via a transformation!

**Comment by pgavlin** [comment]
> I'll give that a go in the meantime, but it would obviously be better to have a first-class way to remove resources via a transformation!

FWIW, that capability is tracked via https://github.com/pulumi/pulumi-kubernetes/issues/486

**Comment by gitfool** [comment]
I used `kubectl` to run the equivalent Grafana test and it worked:
```bash
kubectl exec -it -n monitoring -c grafana po-grafana-56fd4bc598-lb94k bash
curl -s -o /dev/null -I -w '%{http_code}' http://po-grafana/api/health
200
```
... so I'm still not sure why it's always failing.

Meanwhile, I'm very pleased to say the hack to remove resources in a transformation works for me:
```ts
function removeGrafanaTest(obj: any) {
    if (obj.metadata.name === "po-grafana-test") {
        obj.apiVersion = "v1";
        obj.kind = "List";
        obj.items = [];
    }
}
```

**Comment by lblackstone** [comment]
Looks like this was fixed in #486
